### DISCUSSION TITLE
GPU works only first time

### DISCUSSION THREAD

**Discussion by Atefeh197** [openai/whisper#1327] [General]
Hello 

I use the whisper model for English audio files, the first time I used the whisper with GPU, it worked properly, but after that, it had a memory error.

"Cuda is out of memory"!

This is my code:

model = whisper.load_model('large-v2','cpu')
result = model.transcribe("1.wav")

How I can fix the error?

**Comment by phineas-pta** [comment]
you can try to free up gpu memory: `torch.cuda.empty_cache()`

**Reply by Atefeh197** [reply]
Thank you.
I used it, but it does not help me. I also get "cuda out of memory"!

**Reply by phineas-pta** [reply]
seems like you don't have lots of VRAM, i think you have to reload the model after each usage, or use the `medium` model

**Reply by Atefeh197** [reply]
My memory is 64GiB.

I reload the model after each usage.

it is weird because the first time it worked with the large model!

I do not understand what is the problem.

**Reply by Atefeh197** [reply]
This is my error:

torch.cuda.OutOfMemoryError: CUDA out of memory. Tried to allocate 26.00 MiB (GPU 0; 7.92 GiB total capacity; 6.40 GiB already allocated; 44.00 MiB free; 6.74 GiB reserved in total by PyTorch) If reserved memory is >> allocated memory try setting max_split_size_mb to avoid fragmentation.  See documentation for Memory Management and PYTORCH_CUDA_ALLOC_CONF

**Reply by phineas-pta** [reply]
you only have 8 GB VRAM and that's not enough for large model

**Comment by jongwook** [comment]
Looks like the your VRAM (8GB) is not big enough to keep running the `large-v2` model using the implementation in this repo, which is not very well optimized for memory. You may have a better chance with other implementations like [whisper-jax](https://github.com/sanchit-gandhi/whisper-jax) or [faster-whisper](https://github.com/guillaumekln/faster-whisper); the latter seems to require only ~5GB memory for running `large-v2`.
